otx/algorithms/common/adapters/mmcv/hooks.py

```diff
@@ -666,3 +666,84 @@ class ForceTrainModeHook(Hook):
     def before_train_epoch(self, runner):
         """Make sure to put a model in a training mode before train epoch."""
         runner.model.train()
+
+
+@HOOKS.register_module()
+class TwoCropTransformHook(Hook):
```

Comment:
> Just for curiosity, as far as I remember, the similar function was also implemented for Self-SL classification. Is it wrong? I just want to avoid the duplicate implementation

This hook is to handle interval (: when supcon loss is applied) in `TwoCropTransform` implemented in `otx/algorithms/segmentation/adapters/mmseg/data/pipelines.py` and this `TwoCropTransform` is similar with the other one implemented by adria in `otx/mpa/modules/datasets/pipelines/transforms/twocrop_transform.py`. There are obvious differences between both modules, but it is possible to combine them to the single module. But, this process is not a scope of this PR and it can be proceeded when merging mpa to otx.

Reply:
> Could you convert the doc-string format as Google's one? We'll use this format as standard doc-string format
> 
> Ref. https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

But other modules also use the same docstring format. I can modify it as you said, but it is not a scope of this PR and I think it would be better to modify it with other modules. What do you think?

Reply:
> > Could you convert the doc-string format as Google's one? We'll use this format as standard doc-string format
> > Ref. https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
> 
> But other modules also use the same docstring format. I can modify it as you said, but it is not a scope of this PR and I think it would be better to modify it with other modules. What do you think?

We're currently focusing on adding type-hint and doc-string. Although pre-existed code didn't include proper type-hint and doc-string, we suggested adding them to new PRs. So, please do not consider pre-existed code and modify just yours.

Reply:
Done.